### PR TITLE
Feature Tracking: Add support for maps and lists

### PR DESCRIPTION
**Pull request by avilevy18**
## Description

Feature tracking can now support maps and lists.

## Related issue

[b/258211839](http://b/258211839)

## How has this been tested?

Unit tests and integration tests.
## Checklist:
- Unit tests
  - [ ] Unit tests do not apply.
  - [x] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [x] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [ ] This PR introduces no new features.
  - [x] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

**Comment by qingling128** [comment]
Adding @braydonk to help share the review load

File: confgenerator/feature_tracking.go [hunk 1]
```diff
@@ -141,6 +146,7 @@ func trackedMappedComponents[C Component](module string, kind string, m map[stri
 	return features, nil
 }
 
+// TODO: add time.Duration to auto tracking
 func trackingFeatures(c reflect.Value, m metadata, feature Feature) ([]Feature, error) {
```
Comment (braydonk) [inline confgenerator/feature_tracking.go:150]: Would you be open to renaming `m` to `md`? `m` on its own feels weirder now that tracking `map` is supported, feels like it needs more distinction.

File: confgenerator/feature_tracking.go [hunk 2]
```diff
@@ -221,17 +227,109 @@ func trackingFeatures(c reflect.Value, m metadata, feature Feature) ([]Feature,
...
+				features = append(features, Feature{
+					Module: feature.Module,
+					Kind:   feature.Kind,
+					Type:   feature.Type,
+					Key:    append(feature.Key, m.yamlTag, k, "__key"),
+					Value:  key,
+				})
+			}
+
+			m2 := m.deepCopy()
```
Comment (braydonk) [inline confgenerator/feature_tracking.go:277]: Maybe naming this `mdCopy` would be clearer than `m2`?

**Review by braydonk — COMMENTED** [review]
Mostly LGTM, just some naming and safety stuff.

File: confgenerator/feature_tracking.go [hunk 3]
```diff
@@ -229,26 +226,124 @@ func trackingFeatures(c reflect.Value, m metadata, feature Feature) ([]Feature,
...
+			}
+
+			if err != nil {
+				return nil, err
+			}
+			features = append(features, fs...)
+		}
+
+	case kind == reflect.Slice || kind == reflect.Array:
+
```
Comment (braydonk) [inline confgenerator/feature_tracking.go:301]: nit: don't usually need a newline after `case` statements

**Review by braydonk — APPROVED** [review]
LGTM!

**Pull request merged**